pkg/database/fileNode.go

```diff
@@ -86,63 +88,61 @@ func (db *DataBase) GetFileNodeUid(hash string) (string, error) {
 	return ret["hasNode"][0].Uid, nil
 }
 
-func (db *DataBase) GetFileNodesByFileNode(filenode *service.FileNode) ([]*service.FileNode, error) {
+func (db *DataBase) GetFileNodesByFileNode(filenode *service.FileNode, recursive bool) ([]*service.FileNode, error) {
```

Comment:
Please add documentation for this function and explain what you can do with it and also what not e.g. what happens when provided FileNode has hash and filetype set and why.